### PR TITLE
Fix broken Jenkins tests by changing from a Future in CallActor to a VassalCallActor.

### PR DESCRIPTION
**Pull request by mcovarr**
Supersedes the work on sfrazer_bugs.  Non-Docker tests pass on gce-ces-build, I don't know how to run  Docker tests there from the CLI.  This does not include @kshakir's MVCC database changes as those don't appear to be strictly necessary to solve these problems, though they might be desirable on their own merits.

### CODE REVIEW

**Comment by scottfrazer** [comment]
I volunteer to be first reviewer

File: src/main/scala/cromwell/engine/VassalCallActor.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  def props(backend: Backend, command: String, workflowDescriptor: WorkflowDescriptor, call: Call, callInputs: CallInputs, lookup: ScopedLookupFunction): Props =
+    Props(new VassalCallActor(backend, command, workflowDescriptor, call, callInputs, lookup))
+}
+
+/** Actor to manage the execution of a single call. */
+class VassalCallActor(backend: Backend, command: String, workflowDescriptor: WorkflowDescriptor, call: Call, callInputs: CallInputs, lookup: ScopedLookupFunction) extends Actor with CromwellActor {
+  private val log = Logging(context.system, classOf[VassalCallActor])
+  val tag = s"VassalCallActor [UUID(${workflowDescriptor.shortId}):${call.name}]"
+
+  log.info(s"$tag STARTING $command")
```
Comment (scottfrazer) [inline src/main/scala/cromwell/engine/VassalCallActor.scala]: I think these log messages should be cleaned up or removed.  We definitely don't need the command printed out again.  Maybe something like `$tag Start call '${call.name}'`, and corresponding `$tag Finished/Failed '${call.name}'`

Comment (scottfrazer) [inline src/main/scala/cromwell/engine/VassalCallActor.scala]: Actually, the `${call.name}` is a bit redundant because it's already in the `$tag`

**Comment by scottfrazer** [comment]
Should we also include that issue where the WorkflowActor was not switching to Running state early enough?

File: src/main/scala/cromwell/engine/VassalCallActor.scala (hunk 1 above)
Comment (mcovarr) [inline src/main/scala/cromwell/engine/VassalCallActor.scala]: ok to all the cleanups

**Comment by mcovarr** [comment]
I actually had to remove that change that switched to Running state earlier as it was responsible for one of the two broken tests in SimpleWorkflowActorSpec.  `startRunnableCalls` returns a "next state", but this is ignored from the context of the `onTransition` handler.  As a result in one case we were staying in Running when we should have gone to Failed.

**Comment by scottfrazer** [comment]
I think besides the log messages this is good.  @cjllanwarne is the second reviewer

File: src/main/scala/cromwell/engine/CallActor.scala [hunk 2]
```diff
@@ -33,6 +33,10 @@ class CallActor(call: Call, locallyQualifiedInputs: Map[String, WdlValue], backe
 
   override def receive = LoggingReceive {
     case CallActor.Start => handleStart()
+    case VassalCallActor.Finished(outputs) => context.parent ! WorkflowActor.CallCompleted(call, outputs)
```
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/CallActor.scala]: While the name amuses me, perhaps `CallExecutionActor`?

File: src/main/scala/cromwell/engine/VassalCallActor.scala [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    case Failure(e) =>
+      log.error(s"$tag FAILING $command")
+      context.parent ! VassalCallActor.Failed(e)
+  }
+
+  override def receive = LoggingReceive {
+    case badMessage =>
+      val diagnostic = s"$tag: unexpected message $badMessage."
+      log.error(diagnostic)
+      context.parent ! engine.workflow.WorkflowActor.CallFailed(call, diagnostic)
```
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/VassalCallActor.scala]: Do we really want to fail the call if it receives a spurious message? 

It does seem like behavior we'd want to know about and admittedly just logging messages is unlikely to ever be spotted, so perhaps so. OTOH it's not actually affecting the call.

Comment (scottfrazer) [inline src/main/scala/cromwell/engine/VassalCallActor.scala]: I agree that it probably shouldn't fail in this case.  Also if we were to keep it, it should probably send a `VassalCallActor.Failed(new Exception...)`

File: src/main/scala/cromwell/engine/CallActor.scala (hunk 2 above)
Comment (cjllanwarne) [inline src/main/scala/cromwell/engine/CallActor.scala]: Shouldn't this be CallActor.VassalFinished(outputs)?

Comment (geoffjentry) [inline src/main/scala/cromwell/engine/CallActor.scala]: Good catch but other way around. `CallActor.Start` should be reversed, the standard we've been using is the received message is owned by the actor

Comment (geoffjentry) [inline src/main/scala/cromwell/engine/CallActor.scala]: @cjllanwarne was right, I was thinking this was `VassalActor`

File: src/main/scala/cromwell/engine/CallExecutionActor.scala [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      log.info(s"$tag: successful execution.")
+      context.parent ! CallActor.ExecutionFinished(callOutputs)
+    case Failure(e) =>
+      log.error(s"$tag: failed execution.")
+      context.parent ! CallActor.ExecutionFailed(e)
+  }
+
+  /**
+   * The `CallExecutionActor` is not meant to respond to messages, it just starts up, does its work,
+   * and shuts down.
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/CallExecutionActor.scala]: well actually it doesn't shut down, but it probably should

**Comment by cjllanwarne** [comment]
Looks good.

**Pull request merged**